contracts_lib_py/nft_upgradeable.py

```diff
@@ -0,0 +1,142 @@
+from web3 import Web3
+
+from contracts_lib_py.contract_base import ContractBase
+
+
+class NFTUpgradeable(ContractBase):
+    """Class representing the NFTUpgradeable contract."""
+    CONTRACT_NAME = 'NFTUpgradeable'
+
+    def is_approved_for_all(self, account_address, operator_address):
+        """
+        Retrieves if the operator is approved for an account adress
+
+        :param account_address: Account address, str
+        :param operator_address: Account address, str
+        :return: bool
+        """
+        return self.contract.caller.isApprovedForAll(
+            Web3.toChecksumAddress(account_address),
+            Web3.toChecksumAddress(operator_address)
+        )
+
+    def set_proxy_approval(self, operator_address, approved, from_account):
+        """
+        Configure proxy approval for a specific operator address
+
+        :param operator_address: Address of the operator, str
+        :param approved: is approved, bool
+        :param from_account: Sender account, Account
+        :return: bool
+        """
+        tx_hash = self.send_transaction(
+            'setProxyApproval',
+            (Web3.toChecksumAddress(operator_address),
+             approved),
+            transact={'from': from_account.address,
+                      'passphrase': from_account.password,
+                      'keyfile': from_account.key_file}
+        )
+        return self.is_tx_successful(tx_hash)
+
+    def proxy_set_approval_for_all(self, account_address, operator_address, approved, from_account):
```

Comment:
what is the difference between these 2 methods? and can we improve the documentation with that?